dao/FashionKnex.js

```diff
@@ -0,0 +1,89 @@
+const knex = require('knex')
+const Model = require('fashion-model/Model')
+const conflogger = require('conflogger')
+const _getTableName = require('./util/getTableName')
+
+function _clean (document) {
+  return Model.isModel(document)
+    ? Model.clean(document)
+    : document
+}
+
+class FashionKnex {
+  constructor (options) {
+    let {
+      modelType,
+      tableName,
+      logger,
+      knexConnection,
+      knexConfig
+    } = options
+
+    this._logger = conflogger.configure(logger)
+    this._modelType = modelType
+    this._tableName = tableName || _getTableName(modelType)
+    this._knex = knexConnection || knex(knexConfig)
+  }
+
+  async insert (data, returning) {
+    data = _clean(data)
+    returning = returning || '*'
+
+    let result
+
+    try {
+      result = await this._knex
+        .insert(data, 'id')
+        .into(this._tableName)
+        .returning(returning)
+    } catch (err) {
+      const stringified = JSON.stringify(data)
+      this._logger.error(`Error inserting data "${stringified}"`, err)
+      throw err
+    }
+    return result
+  }
+
+  async findById (id, toReturn) {
+    const tableName = this._tableName
+
+    toReturn = toReturn || '*'
+    let wrapped
+    let result
+
+    try {
+      result = await this._knex
+        .select(toReturn)
+        .from(tableName)
+        .where('id', id)
+        .limit(1)
+    } catch (err) {
+      this._logger.error(`Error fetching id "${id}" from  table "${tableName}"`, err)
+      throw err
+    }
+
+    let errors = []
+
+    wrapped = this._modelType.wrap(result[0], errors)
```

Comment:
Hey, good question. I'm not a huge fan of the API, but fashion-model does not throw an error if properties have invalid types upon wrapping. If an array is passed as the second argument, fashion-model will push all errors upon wrapping to that array. The idea is that, there may be some instances where you may want to ignore properties that are wrapped with invalid types and continue with the process. Kinda weird, but the code is behaving correctly as I wrote it. It also may not be documented. I would honestly prefer an option to ignore errors instead the errors array. It's something we could consider implementing.